***Note***: *this report is accompanied by "rating.csv" file, which should be placed in same directory as this file for reproducibility.*<br>
***Initial Actions***:*To create a reproducible document containing our analysis for the  quarterback ratings for the 2008 NFL season (The Sports Network) dataset.*

## Linear Regression from scratch in  R

### <span style="color: navy ">Objective:To create a Least Square model to relate the dependent variable __quarterback rating__ to the __percentage of completions__ (Pct Comp).</span>

   _**Assumption**: we are considering only pct Comp as independent variable only for experimental purpose, there is no real relation between dependent and independent variable_

In [1]:
library(pracma)
d <- read.csv("data/rating.csv") #to read data form csv file
head(d)

First  Last       Team Att Comp Pct.Comp Yds  Yds.per.Att TD Pct.TD Long Int
1 Philip Rivers     SD   478 312  65.3     4009 8.39        34 7.1    67   11 
2 Chad   Pennington MIA  476 321  67.4     3653 7.67        19 4.0    80    7 
3 Kurt   Warner     ARI  598 401  67.1     4583 7.66        30 5.0    79   14 
4 Drew   Brees      NO   635 413  65.0     5069 7.98        34 5.4    84   17 
5 Peyton Manning    IND  555 371  66.8     4002 7.21        27 4.9    75   12 
6 Aaron  Rodgers    GB   536 341  63.6     4038 7.53        28 5.2    71   13 
  Pct.Int Rating.Pts
1 2.3     105.5     
2 1.5      97.4     
3 2.3      96.9     
4 2.7      96.2     
5 2.2      95.0     
6 2.4      93.8

General Linear Equation is:
$y = \beta_0 +\beta_1x_1$

Linear Equation to predict Rating.Pts is:
$Rating.Pts = \beta_0 +\beta_1Pct.Comp$

In [2]:
intercept <- rep(1,each = 32) #took 32 as the no of rows in dataframe is 32
d$intercept <- intercept

In [3]:
library(MASS)
X <- cbind(d$intercept,d$Pct.Comp) #colomn binding intercept and independent column
y <- d$Rating.Pts
beta <- ginv(X)%*% y  #matrix multipication/ dot product
beta

[,1]      
[1,] -22.033064
[2,]   1.745626

Least Square model to relate the dependent variable quarterback rating to the percentage of
completions is: 

$$Rating.Pts = -22.033064 + 1.745626Pct.Comp$$

## Multiple Linear Regression from Scratch R


### <span style="color: navy ">Objective: Create a Least Square model to relate the dependent variable quarterback rating to the percentage of Completions (Pct Comp) and interceptions. (Pct Int)</span>

_**Assumption**: we are considering only pct Comp and pct Int as independent variables only for experimental purpose, there is no real relation between dependent and independent variable_

General Multiple Linear Equation is: $y = \beta_0 +\beta_1x_1 + \beta_2x_2$

Linear Equation to predict Rating.Pts is: $Rating.Pts = \beta_0 +\beta_1Pct.Comp + \beta_2Pct.Int$

In [4]:
X1 <- cbind(d$intercept,d$Pct.Comp,d$Pct.Int)
y1 <- d$Rating.Pts
beta1 <- ginv(X1)%*% y1
beta1

[,1]     
[1,] -9.109043
[2,]  1.662173
[3,] -3.076368

Least Square model to relate the dependent variable quarterback rating to the percentage of
Completions (Pct Comp) and interceptions is:  

$$Rating.Pts = -9.109043 + 1.662173Pct.Comp - 3.076368Pct.Int $$

### <span style="color: navy ">Objective: Determine the least square error (square root of sum of square errors) for models in part a and b.</span>

Least Square error for the model 1(part a) is: 

In [5]:
err = norm(X %*% beta - y,'2')
err

[1] 29.58189

Least Square error for the model 2(part b) is:

In [6]:
err1 = norm(X1 %*% beta1 - y1,'2')
err1

[1] 26.58117

The least square error reduced from 29.58 to 26.58. Hence, we can conclude that the accuracy of the model increased by using an extra variable, namely percentage of interceptions.

### <span style="color: navy ">Objective: Use your LR and MLR models to predict the rating for a quarterback with percentage of completions of 60%, and percentage of interceptions of 3%.</span>


In [7]:
y_hat = c(1,60) %*% beta
y_hat

[,1]    
[1,] 82.70451

The rating for a quarterback with percentage of completions of 60% is ***82.70451</span>***.

In [8]:
y_hat1 = c(1,60,3) %*% beta1
y_hat1

[,1]    
[1,] 81.39223

The rating for a quarterback with percentage of completions
of 60%, and percentage of interceptions of 3% is ***81.39223</span>***.

#### <span style="color: navy ">Go Seahawks!</span>